### DISCUSSION TITLE
homekit controller + tado, ogni tanto non mantiene i cambi di stato

### DISCUSSION THREAD

**Discussion by 50494554524F** [indomus/forum#1616] [HUB: Home Assistant]
premetto che non riesco a replicare il problema visto che capita in maniera completamente casuale; 
internet pare ci sia, non dipende da orario o aggiornamenti etc... ho cercato un po', niente. 
quindi, vorrei sapere se capita anche ad altri.

problema: generic thermostat che pilota una entità climate/valvola di homekit controller/tado.
il termostato invia il comando alla valvola, le entità si aggiornano, tutto ok. 
dopo pochi secondi l'entità climate torna allo stato precedente. parte un loop infinito... devo riavviare il bridge.
sarebbe poco male, ma la caldaia si accende e spegne in continuazione visto che uso lo stato delle valvole...

in questi ultimi giorni mi è già capitato quattro volte e devo venirne a capo...
idee per risolvere?! bug?! devo pensare a qualche cosa per intercettare il problema?!

grazie

**Comment by 50494554524F** [comment]
per adesso ho risolto controllando i cambi di stato ravvicinati e "disabilitando" la valvola

**Reply by Scialla** [reply]
i cambi di stato non avvengono da soli, mi sa che l'unico che può aiutarti sei tu, devi debuggare, guardare lo storico dei cambi di stato della valvola e indagare su chi lo ha cambiato, se va in loop sicuramente una automazione che hai scritto e che è sbagliata, trova l'automazione e fai il debug della stessa (deve avere un id univoco per poterla debuggare)

**Reply by 50494554524F** [reply]
le automazioni vanno in loop perché lo stato cambia da solo... se tolgo le automazioni e cambio lo stato manualmente succede la stessa cosa, torna quello di prima

**Reply by Scialla** [reply]
si ma nello storico vedi chi o cosa ha fatto cambiare lo stato

**Reply by 50494554524F** [reply]
riformulo... ho l'entità climate, non ci sono automazioni attive, da strumenti sviluppatori inverto lo stato, dopo pochi secondi torna sullo stato precedente... nel log vedo solo "nome valvola" + on/off...

**Reply by 50494554524F** [reply]
> per adesso ho risolto controllando i cambi di stato ravvicinati e "disabilitando" la valvola

dopo varie prove devo dire che è abbastanza pulita come soluzione e può servire anche ad altro... esempio di turn_on dello switch wrapper sulla valvola tado

```
          turn_on:
                  
            - variables:
            
                interval: "{{  now() | as_timestamp - states[ 'switch.hall_radiator_valve' ].last_changed | as_timestamp  }}"
          
            - service: input_select.select_option
          
              target:
            
                entity_id: input_select.hall_radiator_valve_mode
                
              data:
              
                option: >-
                
                  {{
                  
                      states( 'input_select.hall_radiator_valve_mode' )
                      
                          if interval | int > 60
                          
                          else
                          
                      'locked'
                  
                  }}


            - condition: state
            
              entity_id: input_select.hall_radiator_valve_mode
           
              state: 
              
                - 'enabled'
                
                - 'demo'

            - service: switch.turn_on
          
              target:
            
                entity_id: >-
                
                  {%  if is_state( 'input_select.hall_radiator_valve_mode', 'enabled' )  %}
                  
                      switch.tado_local_radiator_thermostat_va00000000_wrapper
                      
                  {%  else  %}
                  
                      switch.tado_local_radiator_thermostat_va00000000_wrapper_virtual
                  
                  {%  endif  %}

```
con un pulsante si può modificare mode e riportarlo su enabled per sbloccarla...
